openapi-bal-service/src/main/java/io/ballerina/openapi/converter/service/OpenAPIComponentMapper.java

```diff
@@ -147,10 +147,18 @@ public void createComponentSchema(Map<String, Schema> schema, TypeSymbol typeSym
                 components.setSchemas(schema);
                 break;
             case UNION:
-                Schema unionSchema = handleUnionType((UnionTypeSymbol) type, new Schema<>(), componentName);
-                schema.put(componentName, unionSchema.description(typeDoc));
-                components.setSchemas(schema);
-                break;
+                if (typeRef.definition() instanceof EnumSymbol) {
+                    EnumSymbol enumSymbol = (EnumSymbol) typeRef.definition();
+                    Schema enumSchema = mapEnumValues(enumSymbol);
+                    schema.put(componentName, enumSchema.description(typeDoc));
+                    components.setSchemas(schema);
+                    break;
+                } else {
+                    Schema unionSchema = handleUnionType((UnionTypeSymbol) type, new Schema<>(), componentName);
+                    schema.put(componentName, unionSchema.description(typeDoc));
+                    components.setSchemas(schema);
+                    break;
```

Comment:
shall we have a single `break`statement outside the if else block, instead of having two?